### ISSUE TITLE
NatFeats USB: Transfer size of bulk transfers not returned to Atari side

### ISSUE THREAD

**Issue by czietz** [aranym/aranym#95]
The NatFeats call `USBHOST_SUBMIT_BULK_MSG` returns the return value of `libusb_bulk_transfer()` but not the number of bytes transferred:
https://github.com/aranym/aranym/blob/880e24dd24aaf1579668245fe3bc91f8316010f7/src/natfeat/usbhost.cpp#L841-L846

However, the FreeMiNT driver uses this return value to populate `dev->act_len`, the actual transfer length:
https://github.com/freemint/freemint/blob/f42d8b22ebc68a60377eebfc7d7f60ccf77c823c/sys/usb/src.km/ucd/aranym/aranym-hcd.c#L225-L244

As the [libusb documentation](https://libusb.sourceforge.io/api-1.0/group__libusb__syncio.html#ga2f90957ccc1285475ae96ad2ceb1f58c) explains, the return value can be (among others)
> 0 on success (and populates transferred) 
    [LIBUSB_ERROR_TIMEOUT](https://libusb.sourceforge.io/api-1.0/group__libusb__misc.html#ggab2323aa0f04bc22038e7e1740b2f29efa9b000ac2e6dc4ac864c59579dd08c8bd) if the transfer timed out (and populates transferred)

... which means that `dev->act_len` is filled incorrectly.

Imho, the NatFeats call should return `transferred` in these two cases and and a (negative) error in all other cases to the FreeMiNT driver.

**Comment by DavidGZ** [comment]
@czietz you're right, this must be fixed, probably the confusion came from that libusb_control_transfer() function actually returns     the number of bytes transferred on success.

Thanks for spotting this.

**Comment by th-otto** [comment]
Hm, but when libusb always populates *transferred, then returning a single value from the call might not be good enough.

**Comment by DavidGZ** [comment]
This has been fixed with 88f4713fe0199209bb4dc2f651092d2de3fccaee

Thanks @czietz for pointing to this bug.
